src/Core/OrganizationFeatures/AuthorizationHandlers/GroupAuthHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Bit.Core.Context;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Bit.Core.OrganizationFeatures.AuthorizationHandlers;
+
+class GroupAuthHandler : AuthorizationHandler<GroupOperationRequirement, Group>
+{
+    private readonly ICurrentContext _currentContext;
+
+    public GroupAuthHandler(ICurrentContext currentContext)
+    {
+        _currentContext = currentContext;
+    }
+
+    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
+        GroupOperationRequirement requirement,
+        Group resource)
+    {
```

Comment:
Just like GitHub is warning, this method doesn't need to be async.

Reply:
It's an override, it does need to return a Task, but that means you should just `return Task.CompletedTask` at the end.

However, I just found this interesting blog post: https://dev.to/asik/dont-return-taskfromresult-or-taskcompletedtask-4gcp. 

<details>
<summary>
I wrote up a quick test and it's true! 
</summary>

we should keep this `async` and `await Task.CompletedTask`. Who knew?

```csharp
    private async Task bar(bool error)
    {
        if (error)
        {
            throw new Exception();
        }

        await Task.CompletedTask;
    }

    [Fact]
    public async Task foo()
    {
        var toAwait = bar(error: false);
        var toAwait2 = bar(error: true);

        try
        {
            await toAwait;
            await toAwait2;
        }
        catch
        {
            // caught
        }
    }

```
</details>

```csharp
    protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context,
        GroupOperationRequirement requirement,
        Group resource)
    {
        // Currently all GroupOperationRequirements have the same permission requirements
        // TODO: providers need to be included in the claims
        var org = _currentContext.GetOrganization(resource.OrganizationId);
        var canAccess = org.Type == OrganizationUserType.Owner ||
                        org.Type == OrganizationUserType.Admin ||
                        (org.Permissions?.ManageGroups ?? false);

        if (canAccess)
        {
            context.Succeed(requirement);
        }
        await Task.CompletedTask;
    }
```

Reply:
I've used `await Task.CompletedTask` at the end of each `AuthorizationHandler.HandleRequirementAsync` to cover this. Thanks both for the info! I certainly didn't know about this behaviour, seems quite subtle to me.